x-pack/filebeat/input/awss3/decoding.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package awss3
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/elastic/beats/v7/x-pack/libbeat/reader/parquet"
+)
+
+const (
+	codecParquet = "parquet"
+)
+
+// decoder is an interface for decoding data from an io reader.
+type decoder interface {
+	// decode reads and decodes data from an io reader based on the codec type.
+	decode() error
+}
+
+// newDecoder creates a new decoder based on the codec type.
+// It returns a decoder type and an error if the codec type is not supported.
+// If the reader config codec option is not set, it returns a nil decoder and nil error.
+func newDecoder(p *s3ObjectProcessor, r io.Reader) (decoder, error) {
+	switch p.readerConfig.Decoding.Codec {
+	case "":
+		return nil, nil
```

Comment:
[Suggestion]

Return a no-op decoder here, that way the caller of `newDecoder` only needs to check the error, if there is no error, the returned decoder is guaranteed to work.

Reply:
Right now we require the nil value to branch on legacy paths, so will do this once all the legacy paths are removed.